### PR TITLE
Minimally scope permissions in GitHub Actions workflows

### PR DESCRIPTION
**Pull request by desrosj**
This updates the GitHub Actions workflow files to:

- Grant minimally-scoped permissions to each job to adhere to the principle of least privilege
- Specify a timeout on each job to prevent runaway processes consuming too many minutes (the default is 360)

Once this PR is merged, [the Settings -> Actions -> Workflow permissions setting](https://github.com/WordPress/wordpress-playground/settings/actions) can be changed by a repo admin to "Read repository contents and packages permissions".

For more information, see [PRESS11-470](https://newfold.atlassian.net/browse/PRESS11-470).

## References

- https://docs.github.com/en/actions/reference/security/secure-use
- https://docs.github.com/en/actions/reference/workflows-and-actions/workflow-syntax#jobsjob_idtimeout-minutes

## Use of AI

Cursor was used with (Claude Opus 4.7 and Composer 2.0 at varying points) to analyze the repository and make the initial changes.

When this PR is marked "ready for review" it means that I have manually reviewed all permissions and timeouts that were changed and made any necessary adjustments.

As a part of the analysis, the following summary was created:

## Status

| Field | Value |
|---|---|
| Workflows scanned | 4 (`/Users/jdesrosiers/Sites/GitHub/Organizations/newfold-labs/wp-module-secure-passwords/.github/workflows/codecoverage-main.yml`, `/Users/jdesrosiers/Sites/GitHub/Organizations/newfold-labs/wp-module-secure-passwords/.github/workflows/brand-plugin-test-playwright.yml`, `/Users/jdesrosiers/Sites/GitHub/Organizations/newfold-labs/wp-module-secure-passwords/.github/workflows/auto-translate.yml`, `/Users/jdesrosiers/Sites/GitHub/Organizations/newfold-labs/wp-module-secure-passwords/.github/workflows/satis-update.yml`) |
| Branch | `add/scoped-workflow-permissions` (created) |
| Permissions commit | `6cb62af` |
| Timeouts commit | `942a3cb` |

## Top-level `permissions: {}`

| Category | Entry |
|---|---|
| Workflows that were missing the top-level `permissions: {}` directive (added in this run): | `/Users/jdesrosiers/Sites/GitHub/Organizations/newfold-labs/wp-module-secure-passwords/.github/workflows/codecoverage-main.yml` |
| Workflows that were missing the top-level `permissions: {}` directive (added in this run): | `/Users/jdesrosiers/Sites/GitHub/Organizations/newfold-labs/wp-module-secure-passwords/.github/workflows/brand-plugin-test-playwright.yml` |
| Workflows that were missing the top-level `permissions: {}` directive (added in this run): | `/Users/jdesrosiers/Sites/GitHub/Organizations/newfold-labs/wp-module-secure-passwords/.github/workflows/satis-update.yml` |
| Workflows that were missing the top-level `permissions: {}` directive (added in this run): | `/Users/jdesrosiers/Sites/GitHub/Organizations/newfold-labs/wp-module-secure-passwords/.github/workflows/auto-translate.yml` |

## Job-level `permissions:` additions

| Workflow file | Summary | Job | Permissions / notes |
|---|---|---|---|
| /Users/jdesrosiers/Sites/GitHub/Organizations/newfold-labs/wp-module-secure-passwords/.github/workflows/codecoverage-main.yml | 1 job was missing a scoped `permissions:` directive | get-repo-name | `permissions: {}` [3] |
| /Users/jdesrosiers/Sites/GitHub/Organizations/newfold-labs/wp-module-secure-passwords/.github/workflows/satis-update.yml | 1 job was missing a scoped `permissions:` directive | webhook | `permissions: {}` [3] |

## Permissions corrections (previously incorrect)

| # | Correction |
|---:|---|
| 1 | `/Users/jdesrosiers/Sites/GitHub/Organizations/newfold-labs/wp-module-secure-passwords/.github/workflows/codecoverage-main.yml` :: workflow root: BEFORE `permissions: contents: read` -> AFTER `# Disable permissions…` + `permissions: {}` -- workflow-level default deny per GitHub Actions permissions model ([workflow `permissions`](https://docs.github.com/en/actions/reference/workflows-and-actions/workflow-syntax#permissions)) -- [3] |
| 2 | `/Users/jdesrosiers/Sites/GitHub/Organizations/newfold-labs/wp-module-secure-passwords/.github/workflows/brand-plugin-test-playwright.yml` :: workflow root: BEFORE `permissions: contents: read` -> AFTER `# Disable permissions…` + `permissions: {}` -- same -- [3] |
| 3 | `/Users/jdesrosiers/Sites/GitHub/Organizations/newfold-labs/wp-module-secure-passwords/.github/workflows/satis-update.yml` :: workflow root: BEFORE (no workflow-level `permissions`) -> AFTER `# Disable permissions…` + `permissions: {}` -- same -- [3] |
| 4 | `/Users/jdesrosiers/Sites/GitHub/Organizations/newfold-labs/wp-module-secure-passwords/.github/workflows/auto-translate.yml` :: workflow root: BEFORE `permissions: {}` without the required preceding comment banner -> AFTER comment banner + `permissions: {}` -- matches audit-required formatting -- [3] |
| 5 | `/Users/jdesrosiers/Sites/GitHub/Organizations/newfold-labs/wp-module-secure-passwords/.github/workflows/codecoverage-main.yml` :: `codecoverage`: BEFORE undifferentiated `contents`/`pull-requests` grants without inline rationales -> AFTER same scopes with aligned trailing comments; `permissions` placed immediately after `uses:` -- documents least-privilege intent for reusable coverage + Pages/PR behavior -- [2] |
| 6 | `/Users/jdesrosiers/Sites/GitHub/Organizations/newfold-labs/wp-module-secure-passwords/.github/workflows/brand-plugin-test-playwright.yml` :: `setup`, `bluehost`, `bluehost-dev`: BEFORE `contents: read` without per-permission comments / `uses:` not immediately followed by `permissions:` -> AFTER `uses:` then `permissions:` with `# Required for default token when invoking reusable workflows that clone repositories.` -- clarifies why `contents: read` remains for private-repo reusable calls -- [2] |
| 7 | `/Users/jdesrosiers/Sites/GitHub/Organizations/newfold-labs/wp-module-secure-passwords/.github/workflows/auto-translate.yml` :: `translate`: BEFORE undifferentiated grants -> AFTER same scopes with aligned trailing comments; `permissions` placed immediately after `uses:` -- documents translator reusable workflow needs -- [2] |

## `timeout-minutes` additions

| Workflow | Job | Minutes / action | Rationale |
|---|---|---|---|
| /Users/jdesrosiers/Sites/GitHub/Organizations/newfold-labs/wp-module-secure-passwords/.github/workflows/codecoverage-main.yml | get-repo-name | `30` | bounds a tiny helper job so a hung runner cannot burn minutes indefinitely. |
| /Users/jdesrosiers/Sites/GitHub/Organizations/newfold-labs/wp-module-secure-passwords/.github/workflows/codecoverage-main.yml | codecoverage | `30` | caps runtime for the reusable multi-PHP coverage workflow while staying generous enough for typical CI use. |
| /Users/jdesrosiers/Sites/GitHub/Organizations/newfold-labs/wp-module-secure-passwords/.github/workflows/brand-plugin-test-playwright.yml | setup | `30` | branch extraction should finish quickly; timeout prevents a stuck job. |
| /Users/jdesrosiers/Sites/GitHub/Organizations/newfold-labs/wp-module-secure-passwords/.github/workflows/brand-plugin-test-playwright.yml | bluehost | `30` | bounds Playwright-driven reusable-plugin testing without altering existing behavior. |
| /Users/jdesrosiers/Sites/GitHub/Organizations/newfold-labs/wp-module-secure-passwords/.github/workflows/brand-plugin-test-playwright.yml | bluehost-dev | `30` | same rationale as `bluehost` for the alternate plugin branch matrix. |
| /Users/jdesrosiers/Sites/GitHub/Organizations/newfold-labs/wp-module-secure-passwords/.github/workflows/auto-translate.yml | translate | `30` | translation reusable jobs should complete well under this cap unless upstream APIs stall. |
| /Users/jdesrosiers/Sites/GitHub/Organizations/newfold-labs/wp-module-secure-passwords/.github/workflows/satis-update.yml | webhook | `30` | repository-dispatch path is short; prevents runaway retries or hangs. |

## Notes / blockers

| # | Note |
|---:|---|
| 1 | Reusable workflows under `newfold-labs/workflows` (`reusable-codecoverage.yml`, `module-plugin-test-playwright.yml`, `reusable-translations.yml`) were not checked out in this run; confirm their documented token usage still matches `contents: write`, `pull-requests: write`, and `contents: read` assumptions—especially if those workflows gain new GitHub API surfaces later. |
| 2 | `peter-evans/repository-dispatch@v1` is driven by `secrets.WEBHOOK_TOKEN`; job-level `permissions: {}` assumes the default `GITHUB_TOKEN` is unused—verify if future steps add checkout/`gh`/GitHub-script usage. |

### CODE REVIEW

File: .github/workflows/satis-update.yml [hunk 1]
```diff
@@ -5,10 +5,16 @@ on:
...
+# Disable permissions for all available scopes by default.
+# Any needed permissions should be configured at the job level.
+permissions: {}
+
 jobs:
   webhook:
     name: Send Webhook
     runs-on: ubuntu-latest
+    permissions: {} # WEBHOOK_TOKEN is used for repository-dispatch; default GITHUB_TOKEN is not used against the API.
+    timeout-minutes: 30
```
Comment (desrosj) [inline .github/workflows/satis-update.yml]:
```suggestion
    timeout-minutes: 5
```

This typically completes in less than 10 seconds.

File: .github/workflows/codecoverage-main.yml [hunk 2]
```diff
@@ -28,10 +31,11 @@ jobs:
...
   codecoverage:
     needs: get-repo-name
-    permissions:
-      contents: write
-      pull-requests: write
     uses: newfold-labs/workflows/.github/workflows/reusable-codecoverage.yml@main
+    permissions:
+      contents: write       # Required for pushing coverage HTML to GitHub Pages.
+      pull-requests: write  # Required for reusable workflow PR annotations and checks.
+    timeout-minutes: 30
```
Comment (desrosj) [inline .github/workflows/codecoverage-main.yml]:
```suggestion
```
It does not seem that including `timeout-minutes` here will result in an invalid workflow file. However, this should just be removed to avoid edge cases because it is not explicitly documented as supported and is silently ignored.

File: .github/workflows/brand-plugin-test-playwright.yml [hunk 3]
```diff
@@ -43,9 +46,10 @@ jobs:
   bluehost-dev:
     name: Bluehost Dev Build and Test Playwright
     needs: setup
-    permissions:
-      contents: read
     uses: newfold-labs/workflows/.github/workflows/module-plugin-test-playwright.yml@main
+    permissions:
+      contents: read # Required for default token when invoking reusable workflows that clone repositories.
+    timeout-minutes: 30
```
Comment (desrosj) [inline .github/workflows/brand-plugin-test-playwright.yml]:
```suggestion
```
It does not seem that including `timeout-minutes` here will result in an invalid workflow file. However, this should just be removed to avoid edge cases because it is not explicitly documented as supported and is silently ignored.

File: .github/workflows/brand-plugin-test-playwright.yml [hunk 4]
```diff
@@ -31,9 +33,10 @@ jobs:
   bluehost:
     name: Bluehost Build and Test Playwright
     needs: setup
-    permissions:
-      contents: read
     uses: newfold-labs/workflows/.github/workflows/module-plugin-test-playwright.yml@main
+    permissions:
+      contents: read # Required for default token when invoking reusable workflows that clone repositories.
+    timeout-minutes: 30
```
Comment (desrosj) [inline .github/workflows/brand-plugin-test-playwright.yml]:
```suggestion
```
It does not seem that including `timeout-minutes` here will result in an invalid workflow file. However, this should just be removed to avoid edge cases because it is not explicitly documented as supported and is silently ignored.

File: .github/workflows/auto-translate.yml [hunk 5]
```diff
@@ -12,15 +12,18 @@ concurrency:
...
   translate:
     name: 'Check and update translations'
-    permissions:
-      contents: write
-      pull-requests: write
     uses: newfold-labs/workflows/.github/workflows/reusable-translations.yml@main
+    permissions:
+      contents: write       # Required for reusable workflow to commit updated translation files.
+      pull-requests: write  # Required for reusable workflow to open or update translation PRs.
+    timeout-minutes: 30
```
Comment (desrosj) [inline .github/workflows/auto-translate.yml]:
```suggestion
```
It does not seem that including `timeout-minutes` here will result in an invalid workflow file. However, this should just be removed to avoid edge cases because it is not explicitly documented as supported and is silently ignored.

File: .github/workflows/auto-translate.yml [hunk 6]
```diff
@@ -12,15 +12,18 @@ concurrency:
...
 jobs:
   translate:
     name: 'Check and update translations'
-    permissions:
-      contents: write
-      pull-requests: write
     uses: newfold-labs/workflows/.github/workflows/reusable-translations.yml@main
+    permissions:
+      contents: write       # Required for reusable workflow to commit updated translation files.
+      pull-requests: write  # Required for reusable workflow to open or update translation PRs.
```
Comment (desrosj) [inline .github/workflows/auto-translate.yml]:
```suggestion
      contents: write      # Required for reusable workflow to commit updated translation files.
      pull-requests: write # Required for reusable workflow to open or update translation PRs.
```

File: .github/workflows/brand-plugin-test-playwright.yml [hunk 7]
```diff
@@ -6,19 +6,21 @@ on:
...
+# Any needed permissions should be configured at the job level.
+permissions: {}
+
 jobs:
   setup:
     name: Setup
     runs-on: ubuntu-latest
     permissions:
-      contents: read
+      contents: read # Required for default token when invoking reusable workflows that clone repositories.
```
Comment (desrosj) [inline .github/workflows/brand-plugin-test-playwright.yml]:
```suggestion
      contents: read
```
Don't need to explain why `read` permissions are required.

File: .github/workflows/brand-plugin-test-playwright.yml [hunk 8]
```diff
@@ -31,9 +33,10 @@ jobs:
   bluehost:
     name: Bluehost Build and Test Playwright
     needs: setup
-    permissions:
-      contents: read
     uses: newfold-labs/workflows/.github/workflows/module-plugin-test-playwright.yml@main
+    permissions:
+      contents: read # Required for default token when invoking reusable workflows that clone repositories.
```
Comment (desrosj) [inline .github/workflows/brand-plugin-test-playwright.yml]:
```suggestion
      contents: read
```
Don't need to explain why `read` permissions are required.

File: .github/workflows/brand-plugin-test-playwright.yml [hunk 9]
```diff
@@ -43,9 +46,10 @@ jobs:
   bluehost-dev:
     name: Bluehost Dev Build and Test Playwright
     needs: setup
-    permissions:
-      contents: read
     uses: newfold-labs/workflows/.github/workflows/module-plugin-test-playwright.yml@main
+    permissions:
+      contents: read # Required for default token when invoking reusable workflows that clone repositories.
```
Comment (desrosj) [inline .github/workflows/brand-plugin-test-playwright.yml]:
```suggestion
      contents: read
```

Don't need to explain why `read` permissions are required.

File: .github/workflows/codecoverage-main.yml [hunk 10]
```diff
@@ -13,12 +13,15 @@ on:
...
-  contents: read
+# Disable permissions for all available scopes by default.
+# Any needed permissions should be configured at the job level.
+permissions: {}
 
 jobs:
   get-repo-name:
     runs-on: ubuntu-latest
+    permissions: {} # No repository checkout or GitHub API access required.
+    timeout-minutes: 30
```
Comment (desrosj) [inline .github/workflows/codecoverage-main.yml]:
```suggestion
    timeout-minutes: 5
```
This typically takes less than 5 seconds.

File: .github/workflows/brand-plugin-test-playwright.yml [hunk 11]
```diff
@@ -6,19 +6,21 @@ on:
...
+# Any needed permissions should be configured at the job level.
+permissions: {}
+
 jobs:
   setup:
     name: Setup
     runs-on: ubuntu-latest
     permissions:
       contents: read
+    timeout-minutes: 30
```
Comment (desrosj) [inline .github/workflows/brand-plugin-test-playwright.yml]:
```suggestion
    timeout-minutes: 5
```

File: .github/workflows/brand-plugin-test-playwright.yml [hunk 12]
```diff
@@ -6,19 +6,21 @@ on:
...
+# Disable permissions for all available scopes by default.
+# Any needed permissions should be configured at the job level.
+permissions: {}
+
 jobs:
   setup:
     name: Setup
     runs-on: ubuntu-latest
     permissions:
       contents: read
```
Comment (desrosj) [inline .github/workflows/brand-plugin-test-playwright.yml]:
```suggestion
permissions: {}
```
